lib/graphlient/adapters/faraday_adapter.rb

```diff
@@ -0,0 +1,40 @@
+require 'faraday'
+require 'faraday_middleware'
+
+module Graphlient
+  module Adapters
+    class FaradayAdapter
+      attr_accessor :url, :headers
+
+      def initialize(url, headers: {})
+        @url = url
+        @headers = headers
+      end
+
+      def execute(document:, operation_name:, variables:, context:)
+        response = conn.post do |req|
+          context[:headers].each { |k, v| req.headers[k] = v } if context[:headers]
```

Comment:
Does this work? `req.headers.merge!(context[:headers] || {})`

Reply:
it should! will modify